src/chains/tests/test_views.py

```diff
@@ -181,6 +189,10 @@ def test_match(self):
                 "value": chain.safe_apps_rpc_uri,
             },
             "block_explorer_uri": chain.block_explorer_uri,
+            "block_explorer_uri_template": {
+                "address": chain.block_explorer_uri_address_template,
+                "tx_hash": chain.block_explorer_uri_tx_hash_template,
```

Comment:
I am wondering what would be the best way express this. The assumption is that there is always a template for `address` and `tx_hash` ? Introducing new types here would be a bit cumbersome, right? This looks a lot like a `HashMap<String, String>`

Reply:
Maybe it would be better to have a `List<UriTemplate>` where `UriTemplate` is: 
```rust
struct UriTemplate {
  type: String
  uri: String
}
```
Sorry if this was already discussed with the clients.

Reply:
For simplicity reasons each template is stored in a column. Giving the example that a new chain is added via the Admin interface how would you see a new template being added? 

Currently it is quite loose... it might not even have any template in the URL although this can be changed: I can add a validator that detects if the field being save has something like `{{template}}`

Reply:
> Maybe it would be better to have a `List<UriTemplate>` where `UriTemplate` is:
> 
> ```rust
> struct UriTemplate {
>   type: String
>   uri: String
> }
> ```
> 
> Sorry if this was already discussed with the clients.

The problem with this is that it does not cover situations where there might be multiple template fields to be replaced 🤔

Reply:
To me, it seems the map serves well for this case, what advantages would a list have?

Reply:
The advantage of using `UriTemplate` is typing. Using simply `HashMap<String, String>` makes me feel a bit insecure as to how clients would differentiate between the entry for `tx_hash` and the entry for `address` . It is just string comparison? That seems a bit error prone. Not sure how strict we need to be here, but I would say types make for better self-documenting code.

Reply:
Admittedly, introducing polymorphism here, would make it so that adding a new type, is a breaking change of the API whereas a new entry in the `HashMap<String, String>` is easy. I would just be concerned about the time we need to `HashMap<(String, String, ..., String) , String>` and so on

Reply:
Is it possible to define constants type for keys in the hashmap?

Reply:
That could also be an alternative to have the keys as an enum, but then the variants in the enum would also have to match the string for the path variable name here: `{{placeholder}}`